modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -63,10 +63,23 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
     L: RegistryLookup[F],
     C: Clock[F]
   ): F[Either[ResolutionError, SchemaLookupResult]] = {
-    val get: Registry => F[Either[RegistryError, Json]] = r => L.lookup(r, schemaKey)
+    val get: Registry => F[Either[RegistryError, (Json, Option[SchemaVer.Full])]] = r =>
+      EitherT(L.lookup(r, schemaKey)).semiflatMap { schema =>
+        schema.hcursor.downField("$supersededBy").as[SchemaVer.Full] match {
```

Comment:
Consider returning an error for the malformed  `$supersededBy` value, if it exists. e.g.`match FailedCursor ...`

Reply:
Should we also validate that superseded version is greater than current?